src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const lines = sourceString.split(';');
+  const result = {};
+
+  lines.forEach((line) => {
+    const trimmedLine = line.trim();
+
+    if (trimmedLine === '' || !trimmedLine.includes(':')) {
+      return;
+    }
 
+    const [property, ...valueParts] = trimmedLine.split(':');
+    const value = valueParts.join(':').trim();
+
+    if (property && value) {
+      result[property.trim()] = value.replace(/;$/, '').trim();
+    }
```

Comment:
Instead of using forEach and an external result object, consider using reduce to directly build up the result object. This would align with the checklist's recommendation to avoid modifying an already existing object and would encapsulate the entire transformation within a single expression.